core/templates/domain/story/editable-story-backend-api.service.spec.ts

```diff
@@ -91,24 +77,25 @@ describe('Editable story backend API service', function() {
         description: 'Skill Description'
       }]
     };
-  }));
+  });
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+  afterEach(() => {
+    httpTestingController.verify();
   });
 
   it('should successfully fetch an existing story from the backend',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect(
-        'GET', '/story_editor_handler/data/storyId').respond(
-        sampleDataResults);
-      EditableStoryBackendApiService.fetchStory('storyId').then(
+      editableStoryBackendApiService.fetchStory('storyId').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/story_editor_handler/data/storyId');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
+
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith({
         story: sampleDataResults.story,
```

Comment:
I think these keys should be exactly like the sampleDataResults keys (ie. snake_case)?

Reply:
Fixed.